libultraship/libultraship/DisplayList.cpp

```diff
@@ -12,20 +12,51 @@ namespace Ship
 		while (reader->GetBaseAddress() % 8 != 0)
 			reader->ReadByte();
 
-		while (true)
+		if (sizeof(uintptr_t) < 8)
 		{
-			uint64_t data = reader->ReadUInt64();
+			while (true)
+			{
+				uint64_t data = reader->ReadUInt64();
 
-			dl->instructions.push_back(data);
-				
-			uint8_t opcode = data >> 24;
+				dl->instructions.push_back(data);
 
-			// These are 128-bit commands, so read an extra 64 bits...
-			if (opcode == G_SETTIMG_OTR || opcode == G_DL_OTR || opcode == G_VTX_OTR || opcode == G_BRANCH_Z_OTR || opcode == G_MARKER || opcode == G_MTX_OTR)
-				dl->instructions.push_back(reader->ReadUInt64());
+				uint8_t opcode = data >> 24;
 
-			if (opcode == G_ENDDL)
-				break;
+				// These are 128-bit commands, so read an extra 64 bits...
+				if (opcode == G_SETTIMG_OTR || opcode == G_DL_OTR || opcode == G_VTX_OTR || opcode == G_BRANCH_Z_OTR || opcode == G_MARKER || opcode == G_MTX_OTR)
+					dl->instructions.push_back(reader->ReadUInt64());
+
+				if (opcode == G_ENDDL)
+					break;
+			}
+		}
+		else
+		{
+			while (true)
+			{
+				uint64_t data = reader->ReadUInt64();
+				uint32_t w0 = (uint32_t)data;
+				uint32_t w1 = (uint32_t)(data >> 32);
+
+				dl->instructions.push_back(w0);
+				dl->instructions.push_back(w1);
+
+				uint8_t opcode = (uint8_t)(w0 >> 24);
+
+				if (opcode == G_SETTIMG_OTR || opcode == G_DL_OTR || opcode == G_VTX_OTR || opcode == G_BRANCH_Z_OTR || opcode == G_MARKER || opcode == G_MTX_OTR)
+				{
+					data = reader->ReadUInt64();
+
+					w0 = (uint32_t)data;
+					w1 = (uint32_t)(data >> 32);
+
+					dl->instructions.push_back(w0);
+					dl->instructions.push_back(w1);
+				}
+
+				if (opcode == G_ENDDL)
+					break;
+			}
```

Comment:
What is the reasoning for having two different implementations here?

Reply:
DLists in RAM were increased in size, yes. But the DLists in the OTR are always 64-bits for each `Gfx`. The in-memory DLists being 128-bits per `Gfx` just means there's more room for 64-bit pointers. I don't think we need to split the 64-bits into two and treat them as two `Gfx`?

Reply:
@NEstelami Good with these display list changes after recent updates?

Reply:
CC @Emill